File: SPUIKitCategory/UIView/UIView+Nib.m
//
//  UIView+Nib.m
//  SPKit
//
//  Created by 李双鹏 on 15/10/04.
//  Copyright (c) 2015 LSP. All rights reserved.
//

#import "UIView+Nib.h"

@implementation UIView (SPNib)

+ (UINib *)loadNib
{
    return [self loadNibNamed:NSStringFromClass([self class])];
}

+ (UINib *)loadNibNamed:(NSString*)nibName
{
    return [self loadNibNamed:nibName bundle:[NSBundle mainBundle]];
}

+ (UINib *)loadNibNamed:(NSString*)nibName bundle:(NSBundle *)bundle
{
    return [UINib nibWithNibName:nibName bundle:bundle];
}



#pragma mark ------ XIB INSPECTABLE ------
- (void)setCornerRadius:(CGFloat)cornerRadius {
    self.layer.cornerRadius = cornerRadius;
}
- (CGFloat)cornerRadius {
    return self.layer.cornerRadius;
}

- (void)setShadowOpacity:(CGFloat)shadowOpacity {
    self.layer.shadowOpacity = shadowOpacity;
}
- (CGFloat)shadowOpacity {
    return self.layer.shadowOpacity;
}

- (void)setShadowRadius:(CGFloat)shadowRadius {
    self.layer.shadowRadius = shadowRadius;
}
- (CGFloat)shadowRadius {
    return self.layer.shadowRadius;
}

- (void)setShadowOffset:(CGSize)shadowOffset {
    self.layer.shadowOffset = shadowOffset;
}
- (CGSize)shadowOffset {
    return self.layer.shadowOffset;
}

- (void)setShadowColor:(UIColor *)shadowColor {
    self.layer.shadowColor = shadowColor.CGColor;
}
- (UIColor *)shadowColor {
    return [UIColor colorWithCGColor:self.layer.shadowColor];
}

- (void)setMaskToBounds:(BOOL)maskToBounds {
    self.layer.masksToBounds = maskToBounds;
}
- (BOOL)maskToBounds {
    return self.layer.masksToBounds;
}

@end

File: Sources/UIView/UIView+Nib.swift
//
//  UIView+Nib.swift
//  SPUIKitCategory module
//
//  Created by Bee on 2020/12/15.
//  Copyright © 2020 LSP. All rights reserved.
//

#if !os(macOS)
import UIKit

extension UIView {
    /**
     *  加载nib
     *
     *  @param nibName nib名字 [默认本类的类名]
     *  @param bundle  nib文件包（目录）[默认mainBundle]
     */
    public
    class func loadNibNamed(_ nibName: String = "", bundle: Bundle = .main) -> UINib {
        let name = nibName.isEmpty ? String(describing: self) : nibName
        return UINib(nibName: name, bundle: bundle)
    }

    
    ///mark --------- XIB INSPECTABLE -----------
    @IBInspectable public
    var corner_Radius: CGFloat {
        get {
            return self.layer.cornerRadius
        }
        set {
            self.layer.cornerRadius = newValue
        }
    }
    
    @IBInspectable public
    var shadow_Opacity: Float {
        get {
            return self.layer.shadowOpacity
        }
        set {
            self.layer.shadowOpacity = newValue
        }
    }
    
    @IBInspectable public
    var shadowRadius: CGFloat {
        get {
            return self.layer.shadowRadius
        }
        set {
            self.layer.shadowRadius = newValue
        }
    }
    
    @IBInspectable public
    var shadow_Color: UIColor? {
        get {
            if let color = self.layer.shadowColor {
                return UIColor(cgColor: color)
            } else {
                return nil
            }
        }
        set {
            self.layer.shadowColor = newValue?.cgColor
        }
    }
    
    @IBInspectable public
    var shadow_Offset: CGSize {
        get {
            return self.layer.shadowOffset
        }
        set {
            self.layer.shadowOffset = newValue
        }
    }

    @IBInspectable public
    var masks_ToBounds: Bool {
        get {
            return self.layer.masksToBounds
        }
        set {
            self.layer.masksToBounds = newValue
        }
    }
}
#endif
